src/backend/utils/init/miscinit.c

```diff
@@ -1693,6 +1694,22 @@ process_shared_preload_libraries(void)
 				   false);
 
 	process_shared_preload_libraries_in_progress = false;
+
+	{
+		struct stat st;
+
+		if (file_read_buffer_modify_hook == NULL)
+		{
+			int file_exists = (stat("data_encryption.key", &st) ? (errno != ENOENT) : true);
+			errno = 0; // reset errno because we expected errno in stat
```

Comment:
It looks that we don't need reset errno since it will be set when error occurs. Right?

Reply:
PostgreSQL does not reset errno before IO syscall.

```
stat('') -> -1, errno=ENOENT
fopen() -> 0, but errno still = ENOENT
```

Reply:
If `fopen()` returns a valid pointer we won't check the `errno`. Only when `fopen()` returns `NULL` we will check `errno` and the `errno` should be set by `fopen()`. Am I right?